packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -1477,3 +1551,53 @@ function legacyCreateInstances(cluster: DatabaseClusterNew, props: DatabaseClust
 function databaseInstanceType(instanceType: ec2.InstanceType) {
   return 'db.' + instanceType.toString();
 }
+
+/**
+ * Validate Performance Insights settings
+ */
+function validatePerformanceInsightsSettings(
+  cluster: DatabaseClusterNew,
+  instance: {
+    nodeId?: string;
+    performanceInsightsEnabled?: boolean;
+    performanceInsightRetention?: PerformanceInsightRetention;
+    performanceInsightEncryptionKey?: kms.IKey;
+  },
+): void {
+  const target = instance.nodeId ? `instance \'${instance.nodeId}\'` : '`instanceProps`';
+
+  // If Performance Insights is enabled on the cluster, the one for each instance will be enabled as well.
+  if (cluster.performanceInsightsEnabled && instance.performanceInsightsEnabled === false) {
+    Annotations.of(cluster).addWarningV2(
+      '@aws-cdk/aws-rds:instancePerformanceInsightsOverridden',
+      `Performance Insights is enabled on cluster '${cluster.node.id}' at cluster level, but disabled for ${target}. `+
+      'However, Performance Insights for this instance will also be automatically enabled if enabled at cluster level.',
```

Comment:
should we just throw an error here to match the bebhaviour with the bottom checks?

Reply:
CFn can also deploy with this configuration. It just means that the instance configuration is overridden by the Aurora layer.

If it can be deployed as the correct behaviour in CFn but not in CDK, it will block valid deployments. This is why we have made it a warning here.

What do you think?

Reply:
That's good to know. I agree with your approach instead if it's deployable.